### PR TITLE
Enable gocritic Linter in golangcli.yml

### PR DESCRIPTION
**Pull request by Shubham82**
Signed-off-by: Shubham82 <shubham.kuchhal@india.nec.com>

_Provide a description of what has been changed_
This PR enable gocritic in .golangcli.yml and remove all the lint errors or warnings which shown on running **golangci-lint run**.

### Checklist
- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [ ] Tests have been added
- [ ] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [ ] Changelog has been updated

PartiallyFixes #1158

### CODE REVIEW

**Comment by Shubham82** [comment]
Hi @zroubalik @ahmelsayed @tomkerkhove @jeffhollan
Could you Please take a look.
Thanks.

File: adapter/main.go [hunk 1]
```diff
@@ -113,7 +120,6 @@ func main() {
 
 	logger.Info(cmd.Message)
 	if err := cmd.Run(wait.NeverStop); err != nil {
```
[the remark this answers is not in the document]
Comment (Shubham82) [reply adapter/main.go:122]: Thanks @zroubalik   for taking look in this.
Just want to know that is this expecting behaviour you have mentioned above 
`if err = cmd.Run(wait.NeverStop); err != nil {`

File: pkg/scalers/azure_blob_scaler.go [hunk 2]
```diff
@@ -91,7 +91,8 @@ func parseAzureBlobMetadata(config *ScalerConfig) (*azureBlobMetadata, kedav1alp
 
 	// If the Use AAD Pod Identity is not present, or set to "none"
 	// then check for connection string
-	if config.PodIdentity == "" || config.PodIdentity == kedav1alpha1.PodIdentityProviderNone {
+	switch {
```
Comment (zroubalik) [inline pkg/scalers/azure_blob_scaler.go:94]: can we rewrite this switch to actually switch over `config.PodIdentity`?

File: pkg/scalers/azure_log_analytics_scaler.go [hunk 3]
```diff
@@ -104,8 +104,8 @@ func NewAzureLogAnalyticsScaler(config *ScalerConfig) (Scaler, error) {
 
 func parseAzureLogAnalyticsMetadata(config *ScalerConfig) (*azureLogAnalyticsMetadata, error) {
 	meta := azureLogAnalyticsMetadata{}
-
-	if config.PodIdentity == "" || config.PodIdentity == kedav1alpha1.PodIdentityProviderNone {
+	switch {
```
Comment (zroubalik) [inline pkg/scalers/azure_log_analytics_scaler.go:107]: can we rewrite this switch to actually switch over config.PodIdentity?

File: pkg/scalers/azure_queue_scaler.go [hunk 4]
```diff
@@ -80,7 +80,8 @@ func parseAzureQueueMetadata(config *ScalerConfig) (*azureQueueMetadata, kedav1a
 
 	// If the Use AAD Pod Identity is not present, or set to "none"
 	// then check for connection string
-	if config.PodIdentity == "" || config.PodIdentity == kedav1alpha1.PodIdentityProviderNone {
+	switch {
```
Comment (zroubalik) [inline pkg/scalers/azure_queue_scaler.go:83]: can we rewrite this switch to actually switch over config.PodIdentity?

File: pkg/scalers/azure_servicebus_scaler.go [hunk 5]
```diff
@@ -102,8 +102,8 @@ func parseAzureServiceBusMetadata(config *ScalerConfig) (*azureServiceBusMetadat
 	if meta.entityType == none {
 		return nil, fmt.Errorf("no service bus entity type set")
 	}
-
-	if config.PodIdentity == "" || config.PodIdentity == kedav1alpha1.PodIdentityProviderNone {
+	switch {
```
Comment (zroubalik) [inline pkg/scalers/azure_servicebus_scaler.go:105]: can we rewrite this switch to actually switch over config.PodIdentity?

File: adapter/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zroubalik) [reply adapter/main.go:122]: yeah

File: pkg/scalers/azure_servicebus_scaler.go (hunk 5 above)
Comment (Shubham82) [reply pkg/scalers/azure_servicebus_scaler.go:105]: I will do this.

**Comment by Shubham82** [comment]
Hi @zroubalik 
I have addressed the comments, PTAL

**Review by zroubalik — APPROVED** [review]
LGTM

**Pull request merged**